lib/active_replicas/rails4/process_local_connection_handler.rb

```diff
@@ -114,6 +122,18 @@ def clear_proxies!
       def current_thread_id
         Thread.current.object_id
       end
+
+      private
+
+      # Sets up `@primary_pool` and `@replica_pools` from the current
+      # `@proxy_configuration`.
+      def initialize_pools
```

Comment:
Not a massive fan of this additional layer of redirection. Any particular motivation for this abstraction?